corehq/form_processor/backends/sql/processor.py

```diff
@@ -145,16 +145,20 @@ def save_processed_models(cls, processed_forms, cases=None, stock_result=None, p
             FormAccessorSQL.save_new_form(processed_forms.submitted)
             if cases:
                 for case in cases:
-                    if toggles.SORT_OUT_OF_ORDER_FORM_SUBMISSIONS_SQL.enabled(
-                            case.domain, toggles.NAMESPACE_DOMAIN):
-                        SqlCaseUpdateStrategy(case).reconcile_transactions_if_necessary()
-
                     CaseAccessorSQL.save_case(case)
 
             if stock_result:
                 ledgers_to_save = stock_result.models_to_save
                 LedgerAccessorSQL.save_ledger_values(ledgers_to_save, stock_result)
 
+        if cases:
+            sort_submissions = toggles.SORT_OUT_OF_ORDER_FORM_SUBMISSIONS_SQL.enabled(
+                processed_forms.submitted.domain, toggles.NAMESPACE_DOMAIN)
+            if sort_submissions:
+                for case in cases:
+                    SqlCaseUpdateStrategy(case).reconcile_transactions_if_necessary()
+                    CaseAccessorSQL.save_case(case)
```

Comment:
Surely we only need to save the case if it actually changed right?

Reply:
https://github.com/dimagi/commcare-hq/pull/22447

Reply:
oops